service/controller/clusterapi/v22/resource/clusterconfigmap/desired.go

```diff
@@ -19,12 +20,26 @@ func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) ([]*cor
 		return nil, microerror.Mask(err)
 	}
 
+	cc, err := controllercontext.FromContext(ctx)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	var workerCount int32
+
+	for _, nodePool := range cc.Status.Worker {
+		workerCount += nodePool.Nodes
+	}
+
 	var configMap *corev1.ConfigMap
 	{
-		v := map[string]string{
+		v := map[string]interface{}{
 			"baseDomain":   key.TenantBaseDomain(cr),
 			"clusterDNSIP": r.dnsIP,
 			"clusterID":    key.ClusterID(&cr),
+			"ingressController": map[string]interface{}{
```

Comment:
Currently we just use `.Values.controller.replicas`. But I think we should structure this a bit more since the configmap is added to all apps for the cluster.

@corest WDYT?

Reply:
Do you mean changing template with `ingressController.replicas`?
For me it makes sense to go even further and have some meaningful root key e.g. apps:
```
apps:
  ingressController:
    replicas: x
```

So that we can keep values better structured. But is that a permanent solution or temporary?

Reply:
The cluster configmap is only for apps so I'm not sure that makes sense.

But maybe the root should be `giantswarm` so it's clear it's something we're managing?

I think this is temporary until we can enable HPA by default but that ticket has been open a long time. 😓

Reply:
Ok, if temporary then I'm fine with this flat structure